deepspeed/runtime/zero/stage3.py

```diff
@@ -141,6 +141,7 @@ def __init__(
         self,
         module,
         init_optimizer,
+        param_names,
```

Comment:
https://github.com/deepspeedai/DeepSpeed/blob/17d80ce4401d4453f6486c22607afb1043edaef1/deepspeed/runtime/zero/stage3.py#L2868-L2872

In earlier versions we passed `param_shapes` to supply parameter names for each subgroup. Could we reuse `param_shapes` here instead of introducing a new `param_names` argument, which seems redundant? Or is there a strong reason to replace `param_shapes` with `param_names`?

Reply:
This is a critical question! 

But No, we can not use `param_shapes`! Because the `param_shapes` is from checkpoint, `param_shapes` comes from the checkpoint, that is from the last distributed environment, that is the `sub group` (or `fp16_group`)  is used in last distributed environment.

If the world size is changed, the `sub group` (or `fp16_group`) will changed. Here we set optimizer parameter for current `sub group` (or `fp16_group`). 

In fact, I also used `param_shapes` at first. But I found this problem when my new unit test (which sub_group_size is set to 100) failed to run.

Reply:
OK, I have remove `param_shapes`.